### PR TITLE
initial docs for create sink

### LINKED ISSUE #219: Document CREATE SINK
### Related code PR

https://github.com/singularity-data/risingwave/pull/3674
https://github.com/singularity-data/risingwave/pull/3923
https://github.com/singularity-data/risingwave/pull/3942
https://github.com/singularity-data/risingwave/pull/3682

### Which part(s) of the docs might be affected or should be updated? And how?

Add a new command: 'CREATE SINK'.
We also need to update the product description to reflect that we already support data sinks.

### Reference

e2e test for sink: https://github.com/singularity-data/risingwave/tree/main/e2e_test/sink

### PR DESCRIPTION
**Pull request by hengm3467**
## Info
- **Description**: 
[ What's changed? ]

Add docs for data sinks, including the syntax, and descriptions in the Introduction article. 

- **Related code PR**: 
https://github.com/singularity-data/risingwave/pull/3674
https://github.com/singularity-data/risingwave/pull/3923
https://github.com/singularity-data/risingwave/pull/3942
https://github.com/singularity-data/risingwave/pull/3682

- **Related doc issue**: 
Resolves https://github.com/singularity-data/risingwave-docs/issues/219

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [x] I have checked the doc site preview and the updated parts look all good. <details><summary>How?</summary>image</details>
  - [ ] (For version-specific PR) I have confirmed that this PR is for a released version. If the software version is not released, put it on hold.

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

File: docs/Introduction.md [hunk 1]
```diff
@@ -6,9 +6,9 @@ sidebar_position: 1
 ---
 
 
-RisingWave is a cloud-native streaming database that uses SQL as the interface. It is designed to reduce the complexity and cost of building real-time applications. RisingWave consumes streaming data, performs continuous queries, and updates results dynamically. As a database system, RisingWave maintains results in its own storage so that users can access data efficiently.
+RisingWave is a cloud-native streaming database that uses SQL as the interface. It is designed to reduce the complexity and cost of building real-time applications. RisingWave consumes streaming data, performs computations when new data come in, and updates results dynamically. As a database system, RisingWave maintains results in its own storage so that users can access data efficiently. You can sink data from RisingWave to an external stream for storage or additional processing.
```
Comment (lmatz) [inline docs/Introduction.md]: Shall we add `incrementally` into:
> performs computations incrementally when new data come in

No strong opinion on this

**Review by lmatz — APPROVED** [review]

File: docs/Introduction.md (hunk 1 above)
Comment (hengm3467) [reply docs/Introduction.md]: @lmatz This is a good suggestion! Let me add "incrementally".

Comment (hengm3467) [reply docs/Introduction.md]: I ended up saying "...performs incremental computations..."

File: docs/sql/commands/sql-create-sink.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+id: create-sink
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-create-sink.md:2]: image
image

- We keep the slug, id, and file name the same for one file.
- The description starts with the base form of a verb.

File: docs/sql/commands/sql-create-sink.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+FROM sink_from
+WITH (
+   connector='kafka',
+   kafka.brokers='broker_address',
+   kafka.topic='topic_address',
+   format='format'
+);
+```
+
+### Parameters
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-create-sink.md:24]: We can drop `### Parameters` in all command references as the table header is self-explanatory. What do you think?
image

File: docs/sql/commands/sql-create-sink.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+|Parameter|	Default|Type|Description|Required?|
+|---|---|---|---|---|
+|sink_name| None|String|Name of the sink to be created| Yes|
+|sink_from| None|String|The source from which data will be output. It can be a materialized source, a materialized view, or a table.|Yes|
+|connector| None|String|Sink connector type. Currently, only 'kafka' is supported.| Yes|
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-create-sink.md:30]: Suggest using a consistent style for the examples/formats ('Kafka', `'ip:port,ip:port'`, `append-only`).
image
I personally prefer `'xxx'`.

File: docs/Introduction.md [hunk 5]
```diff
@@ -6,9 +6,9 @@ sidebar_position: 1
 ---
 
 
-RisingWave is a cloud-native streaming database that uses SQL as the interface. It is designed to reduce the complexity and cost of building real-time applications. RisingWave consumes streaming data, performs continuous queries, and updates results dynamically. As a database system, RisingWave maintains results in its own storage so that users can access data efficiently.
+RisingWave is a cloud-native streaming database that uses SQL as the interface. It is designed to reduce the complexity and cost of building real-time applications. RisingWave consumes streaming data, performs incremental computations when new data come in, and updates results dynamically. As a database system, RisingWave maintains results in its own storage so that users can access data efficiently. You can sink data from RisingWave to an external stream for storage or additional processing.
```
Comment (CharlieSYH) [inline docs/Introduction.md:9]: Google style guide uses singular 'data'.
image

File: docs/sql/commands/sql-create-sink.md (hunk 3 above)
Comment (hengm3467) [reply docs/sql/commands/sql-create-sink.md:24]: good suggestion!

File: docs/Introduction.md (hunk 5 above)
Comment (hengm3467) [reply docs/Introduction.md:9]: Good catch!

File: docs/sql/commands/sql-create-sink.md (hunk 4 above)
Comment (hengm3467) [reply docs/sql/commands/sql-create-sink.md:30]: I think code formatting is enough. Adding an additional layer of style without a clear reason is overkill.

**Pull request merged**